plugins/azure/resoto_plugin_azure/resource/base.py

```diff
@@ -197,6 +200,10 @@ class AzureResourceGroup(AzureResource):
     }
     managed_by: Optional[str] = field(default=None, metadata={'description': 'The id of the resource that manages this resource group.'})  # fmt: skip
     provisioning_state: Optional[str] = field(default=None, metadata={"description": "The resource group properties."})
+    
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        if resource_id := self.id:
```

Comment:
Doesn't this create an edge to itself? Where are the resources in the group fetched?

Reference:
```
GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/resources
```

lists all resources.

It will return a list like this:

```json

{
    "value": [
        {
            "id": "/subscriptions/38b02a39-99c8-45bd-a92e-38b616f109df/resourceGroups/FOO/providers/Microsoft.Compute/disks/test_OsDisk_1_63593392f2674e498ea592919fb12c92",
            "location": "westeurope",
            "managedBy": "/subscriptions/foo/resourceGroups/foo/providers/Microsoft.Compute/virtualMachines/test",
            "name": "test_OsDisk_1_63593392f2674e498ea592919fb12c92",
            "sku": {
                "name": "Premium_LRS",
                "tier": "Premium"
            },
            "type": "Microsoft.Compute/disks"
        },
        {
            "id": "/subscriptions/foo/resourceGroups/foo/providers/Microsoft.Compute/sshPublicKeys/test_key",
            "location": "westeurope",
            "name": "test_key",
            "tags": {},
            "type": "Microsoft.Compute/sshPublicKeys"
        }
   ]
}
```